GenderPayGap.WebUI/Models/Organisation/OrganisationReportsViewModel.cs

```diff
@@ -0,0 +1,156 @@
+                    );
+                    scopesByYear.Add(reportingYear, orgScopeForYear);
+                }
+            }
+
+            return scopesByYear;
+        }
+
+    }
+
+    public class OrganisationScopeForYear
+    {
+
+        private readonly int reportingYear;
+
+        private readonly DateTime snapshotDateForYear;
+
+        private readonly OrganisationScope scopeForYear;
+
+        private readonly Return returnForYear;
+
+        private readonly bool isDraftReturnAvailable;
+
+        public OrganisationScopeForYear(int reportingYear, DateTime snapshotDateForYear, OrganisationScope scopeForYear, Return returnForYear, bool isDraftReturnAvailable)
+        {
+            this.reportingYear = reportingYear;
+            this.snapshotDateForYear = snapshotDateForYear;
+            this.scopeForYear = scopeForYear;
+            this.returnForYear = returnForYear;
+            this.isDraftReturnAvailable = isDraftReturnAvailable;
+        }
+
+        public bool CanChangeScope()
+        {
+            return Math.Abs(reportingYear - VirtualDateTime.Now.Year) <= Global.EditableScopeCount;
+        }
+
+        public string GetScopeVariantText()
+        {
+            if (!scopeForYear.IsInScopeVariant())
+            {
+                return "NOT REQUIRED TO REPORT";
+            }
+
+            return "REQUIRED TO REPORT";
+        }
+
+        public string GetByReportingDeadlineText(DateTime deadline)
+        {
+            if (scopeForYear.IsInScopeVariant())
+            {
+                return "by " + deadline.ToString("d MMM yyyy");
+            }
+
+            return null;
+        }
+
+        public string GetReportStatusText()
+        {
+            if (returnForYear == null)
+            {
+                return "Your organisation has not reported";
+            } 
+            
+            if (returnForYear.IsVoluntarySubmission())
+            {
+                return "Reported voluntarily on " + new GDSDateFormatter(returnForYear.Modified).FullStartDate;
+            }
+
+            return "Reported on " + new GDSDateFormatter(returnForYear.Modified).FullStartDate;
+        }
+
+        public bool IsAnyReportAvailable()
+        {
+            return returnForYear != null || isDraftReturnAvailable;
+        }
+        
+        public string GetReportButtonText()
+        {
+            var isReportAvailable = returnForYear != null;
+
+            if (!isReportAvailable && !isDraftReturnAvailable)
+            {
+                return "Draft report";
+            }
+
+            if (!isReportAvailable && isDraftReturnAvailable)
+            {
+                return "Edit draft";
+            }
+
+            if (isReportAvailable && !isDraftReturnAvailable)
+            {
+                return "Edit report";
+            }
+
+            return "Edit draft report";
+        }
+
+        public DateTime GetSnapshotDate()
+        {
+            return this.snapshotDateForYear;

```

Comment:
Does this need to be here? can you expose a getter on `snapshotDateForYear` directly?